### PR TITLE
Restructured Behov object, based on new insight and discussions

### PR DESCRIPTION
**Pull request by geiralund**

### CODE REVIEW

File: src/main/avro/behov.avdl [hunk 1]
```diff
@@ -21,14 +25,21 @@ protocol Dagpenger {
...
+
+  record HenvendelsesType {
+    union { null, Søknad } søknad = null;
+    union { null, Ettersending } ettersending = null;
+    union { null, Annet } annet = null;
   }
 
   record Behov {
     string behovId;
+    HenvendelsesType henvendelsesType;
```
Comment (geiralund) [inline src/main/avro/behov.avdl]: eller `InnsendingsType`

File: src/main/avro/behov.avdl [hunk 2]
```diff
@@ -1,14 +1,18 @@
 @namespace("no.nav.dagpenger.events.avro")
 protocol Dagpenger {
-  enum JournalpostType {
-      NY, GJENOPPTAK, ETTERSENDING, MANUELL, UKJENT
+
+  record Søknad {
+    union { null, string } vedtakstype = null;
```
Comment (geiralund) [inline src/main/avro/behov.avdl]: We could use enums here (`Ny`, `Gjenopptak`). I do have a bad experience with enums - as they are hard to change when they go out in the wild. But, they could also be very helpful, if they are stable. 

Same with `rettighetsType` (`Ordinær`, `Perm`, `Fisk`.. )

File: src/main/avro/behov.avdl (hunk 1 above)
Comment (nicokk) [reply src/main/avro/behov.avdl]: HenvendelsesType > InnsendingsType imo

File: src/main/avro/behov.avdl [hunk 3]
```diff
@@ -19,16 +23,22 @@ protocol Dagpenger {
...
+
+  record HenvendelsesType {
+    union { null, Søknad } søknad = null;
+    union { null, Ettersending } ettersending = null;
+    union { null, Annet } annet = null;
   }
 
   record Behov {
     string behovId;
+    HenvendelsesType henvendelsesType;
```
Comment (nicokk) [inline src/main/avro/behov.avdl]: Vi trenger også et feilflagg

**Pull request merged**